lighthouse-core/scripts/pptr-run-devtools.js

```diff
@@ -0,0 +1,180 @@
+/**
+ * @license Copyright 2021 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/**
+ * USAGE:
+ * Make sure CHROME_PATH is set to a modern version of Chrome.
+ * This script won't work on older versions that use the "Audits" panel.
+ *
+ * URL list file: yarn run-devtools < path/to/urls.txt
+ * Single URL: yarn run-devtools "https://example.com"
+ */
+
+const puppeteer = require('puppeteer');
+const fs = require('fs');
+const readline = require('readline');
+const yargs = require('yargs/yargs');
+
+/** @typedef {{result?: {value?: string, objectId?: number}, exceptionDetails?: object}} RuntimeEvaluateResponse */
+
+const argv = yargs(process.argv.slice(2))
+  .usage('$0 [url]')
+  .help('h').alias('h', 'help')
+  .option('_', {type: 'string'})
+  .option('o', {
+    type: 'string',
+    default: 'latest-run/devtools-lhrs',
+  }).alias('o', 'output-dir')
+  .option('d', {
+    type: 'string',
+  }).alias('d', 'custom-devtools-frontend')
+  .argv;
+
+/**
+ * https://source.chromium.org/chromium/chromium/src/+/master:third_party/devtools-frontend/src/front_end/test_runner/TestRunner.js;l=170;drc=f59e6de269f4f50bca824f8ca678d5906c7d3dc8
+ * @param {Record<string, function>} receiver
+ * @param {string} methodName
+ * @param {function} override
+ */
+function addSniffer(receiver, methodName, override) {
+  const original = receiver[methodName];
+  if (typeof original !== 'function') {
+    throw new Error('Cannot find method to override: ' + methodName);
+  }
+
+  /**
+   * @param  {...any} args
+   */
+  receiver[methodName] = function(...args) {
+    let result;
+    try {
+      result = original.apply(this, args);
+    } finally {
+      receiver[methodName] = original;
+    }
+    // In case of exception the override won't be called.
+    try {
+      Array.prototype.push.call(args, result);
+      override.apply(this, args);
+    } catch (e) {
+      throw new Error('Exception in overriden method \'' + methodName + '\': ' + e);
+    }
+    return result;
+  };
+}
+
+const sniffLhr = `
+new Promise(resolve => {
+  (${addSniffer.toString()})(
+    UI.panels.lighthouse.__proto__,
+    '_buildReportUI',
+    (lhr, artifacts) => resolve(lhr)
+  );
+});
+`;
+
+const startLighthouse = `
+(async () => {
+  await UI.ViewManager.instance().showView('lighthouse');
```

Comment:
bump on connor's request for ViewManager fallback (though it might not matter at this point?)